src/earn/BeforeDepositBottomSheet.tsx

```diff
@@ -0,0 +1,277 @@
+import { getTokenAnalyticsProps } from 'src/tokens/utils'
+
+function ActionCard({ action }: { action: BeforeDepositAction }) {
+  return (
+    <Touchable
+      style={styles.touchable}
+      key={action.name}
+      borderRadius={20}
+      onPress={action.onPress}
+      testID={`Earn/BeforeDepositBottomSheet/${action.name}`}
+    >
+      <>
+        <action.iconComponent color={Colors.black} />
+        <View style={styles.cardContainer}>
+          <Text style={styles.actionTitle}>{action.title}</Text>
+          <Text style={styles.actionDetails}>{action.details}</Text>
+        </View>
+      </>
+    </Touchable>
+  )
+}
+
+function AddAction({
+  token,
+  forwardedRef,
+}: {
+  token: TokenBalance
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+}) {
+  const { t } = useTranslation()
+
+  const action = {
+    name: BeforeDepositActionName.Add,
+    title: t('addFundsActions.add'),
+    details: t('earnFlow.addCryptoBottomSheet.actionDescriptions.add', {
+      tokenSymbol: token.symbol,
+      tokenNetwork: NETWORK_NAMES[token.networkId],
+    }),
+    iconComponent: QuickActionsAdd,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_before_deposit_action_press, {
+        action: BeforeDepositActionName.Add,
+        ...getTokenAnalyticsProps(token),
+      })
+
+      navigate(Screens.FiatExchangeAmount, {
+        tokenId: token.tokenId,
+        flow: CICOFlow.CashIn,
+        tokenSymbol: token.symbol,
+      })
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+function TransferAction({
+  token,
+  exchanges,
+  forwardedRef,
+}: {
+  token: TokenBalance
+  exchanges: ExternalExchangeProvider[]
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+}) {
+  const { t } = useTranslation()
+
+  const action = {
+    name: BeforeDepositActionName.Transfer,
+    title: t('addFundsActions.transfer'),
+    details: t('earnFlow.addCryptoBottomSheet.actionDescriptions.transfer', {
+      tokenSymbol: token.symbol,
+      tokenNetwork: NETWORK_NAMES[token.networkId],
+    }),
+    iconComponent: QuickActionsSend,
+    onPress: () => {
+      AppAnalytics.track(EarnEvents.earn_before_deposit_action_press, {
+        action: BeforeDepositActionName.Transfer,
+        ...getTokenAnalyticsProps(token),
+      })
+
+      navigate(Screens.ExchangeQR, { flow: CICOFlow.CashIn, exchanges })
+      forwardedRef.current?.close()
+    },
+  }
+  return <ActionCard action={action} />
+}
+
+function CrossChainSwapAction({
+  token,
+  hasTokensOnSameNetwork,
+  forwardedRef,
+}: {
+  token: TokenBalance
+  hasTokensOnSameNetwork: boolean
+  forwardedRef: React.RefObject<BottomSheetModalRefType>
+}) {
+  const { t } = useTranslation()
+
+  const action = {
+    name: BeforeDepositActionName.CrossChainSwap,
+    title: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.action.crossChainSwap')
+      : t('addFundsActions.swap'),
```

Comment:
can we just get all the strings from a single place. E.g., `earnFlow.beforeDepositBottomSheet` over reusing stuff from other places (like `addFundsActions` or `addCryptoBottomSheet`)? I guess some will be duplicated, but it's much more straightforward to change in case we need to in the future